app/scripts/controllers/app-state.js

```diff
@@ -401,6 +404,29 @@ export default class AppStateController extends EventEmitter {
     this.store.updateState({ showAccountBanner });
   }
 
+  /**
+   * Sets an object with networkName and appName
```

Comment:
Hmm, this description doesn't seem totally accurate.

Rather than the network name, what we're getting here is the full network configuration object. And instead of an "app name", we're getting an origin (which is in a property called `siteName`).

Aside from updating this description, perhaps we could substitute the `network` property with the `networkClientId` for the network in question? We can use that to derive all of the metadata we need, rather than storing a second copy of it here, making for a simpler and smaller state shape.